### ISSUE TITLE
pip install MDAnalysis failing on Mac OSX 10.10 Yosemite

### ISSUE THREAD

**Issue by carolynlphillips** [MDAnalysis/mdanalysis#459]
Using pip OR easy_intall the following error is generated

>  src/numtools/calc_distances.h:27:10: fatal error: 'omp.h' file not found

>  #include <omp.h>
>           ^
>  1 warning and 1 error generated.
>
>  error: command '/usr/bin/clang' failed with exit status 1

I note that the build thought that OpenMP was available  (See Below)

Installing collected packages: MDAnalysis

  Running setup.py install for MDAnalysis

    Complete output from command /opt/local/Library/Frameworks/Python.framework/Versions/2.7/Resources/Python.app/Contents/MacOS/Python -c "import setuptools, tokenize;__file__='/private/tmp/pip-build-mq68Zp/MDAnalysis/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record /tmp/pip-Vki164-record/install-record.txt --single-version-externally-managed --compile:

    Attempting to autodetect OpenMP support... Compiler supports OpenMP

**Comment by orbeckst** [comment]
See https://groups.google.com/d/msg/mdnalysis-discussion/yblLSNvucq0/KVzMGUfaBAAJ for background and (manual) workaround.

**Comment by kain88-de** [comment]
actually does our openmp detection work reliable? Because for me the detection algorithm says that my compilers don't support openmp. But the installed clang/llvm(clang-3.5) and gcc(5.2.x) I have both support it.

I tested the [hello world](http://people.sc.fsu.edu/~jburkardt/c_src/hello_openmp/hello_openmp.html) openmp example.

To check the result of our openmp detection use.

```
from setup import has_function
from distutils.ccompiler import new_compiler
cc = new_compiler
print(has_function(cc, 'omp_get_num_threads()')
```

On my machine this returns `False`. That is a false negative. I can run and compile openmp applications.

**Comment by orbeckst** [comment]
Then what should one check for to reliably detect OpenMP? What do `cmake` or `autoconf` use?

It becomes pretty annoying once the user has to explicitly enable it although your #462 makes it simpler:
```
MDA_USE_OPENMP=True  pip install --upgrade MDAnalysis
```

**Comment by kain88-de** [comment]
After experimenting a little bit I got to a version that works on my system.

https://github.com/kain88-de/mdanalysis/blob/b02b97cd9f5ac41ba1ba774005abace714581cbf/package/setup.py#L178

@carolynlphillips you try out my openmp-detection branch? You have to manually clone my repository and checkout the openmp-detection branch

```
git clone https://github.com/kain88-de/mdanalysis
cd mdanalysis
git checkout -b openmp-detection origin/openmp-detection
cd package
python setup.py build_ext --inplace
```

But I'll have a look around what other projects are currently doing and see what the cython-devs suggest.

**Comment by kain88-de** [comment]
We might also copy the update compiler detection from MDTraj. In Debian is detects everything correctly. I still have to check osx

https://github.com/mdtraj/mdtraj/blob/7e3f5b2453d4c9b2e68b61d6f8660da26eac58f1/basesetup.py#L37

**Comment by carolynlphillips** [comment]
This, interestingly, did not work

It did, indeed, determine that OpenMP was not supported... and then it
crashed on omp.h not being available anyway!

> Attempting to autodetect OpenMP support... Did not detect OpenMP support.
....

>In file included from MDAnalysis/lib/distances_openmp.c:256:

*>MDAnalysis/lib/include/calc_distances.h:27:10: **fatal error: **'omp.h'
file not found*

**Comment by kain88-de** [comment]
There is still a logic error from me in the script.  You can try the current version. That should work

**Comment by carolynlphillips** [comment]
Yes.  Now it built successfully.

**Comment by richardjgowers** [comment]
@kain88-de if you want to push this to develop it can be 0.12.1

**Comment by kain88-de** [comment]
If you don't mind I'd like to do some more reading before I commit it to develop.

**Comment by orbeckst** [comment]
Has this issue been fixed with 66d73c607fcbd039a99a8a739c88850f84de1330 ? If so, can someone please confirm and close?

**Comment by kain88-de** [comment]
works for me on El Captain.
